app/fileUploadParsers/Parser.scala

```diff
@@ -16,41 +16,156 @@
+      Field(MemberDetailsFieldNames.nino, ninoField(chargeFields), MemberDetailsFieldNames.nino, 2)
+    )
+    memberDetailsForm
+      .bind(Field.seqToMap(fields))
+      .fold(
+        formWithErrors => Left(errorsFromForm(formWithErrors, fields, index)),
+        value => Right(value)
+      )
+  }
+
+  protected final def errorsFromForm[A](formWithErrors: Form[A], fields: Seq[Field], index: Int): Seq[ParserValidationError] = {
+    formWithErrors
+      .errors
+      .flatMap { formError =>
+        fields.find(_.columnName == formError.key)
+          .map(f => ParserValidationError(index, f.columnNo, formError.message))
+          .toSeq
+      }
+  }
+
+  protected final def combineValidationResults[A, B](
+                                                      resultA: Either[Seq[ParserValidationError], A],
+                                                      resultB: Either[Seq[ParserValidationError], B],
+                                                      resultAJsPath: => JsPath,
+                                                      resultAJsValue: A => JsValue,
+                                                      resultBJsPath: => JsPath,
+                                                      resultBJsValue: => B => JsValue
+                                                    ): Either[Seq[ParserValidationError], Seq[CommitItem]] = {
+    resultA match {
+      case Left(resultAErrors) =>
+        resultB match {
+          case Left(resultBErrors) =>
+            Left(resultAErrors ++ resultBErrors)
+          case _ => Left(resultAErrors)
+        }
 
-  protected def firstNameField(fields: Array[String]):String =fields(0)
-  protected def lastNameField(fields: Array[String]):String =fields(1)
-  protected def ninoField(fields: Array[String]):String =fields(2)
+      case Right(resultAObject) =>
+        resultB match {
+          case Left(resultBErrors) => Left(resultBErrors)
+          case Right(resultBObject) =>
+            Right(
+              Seq(
+                CommitItem(resultAJsPath, resultAJsValue(resultAObject)),
+                CommitItem(resultBJsPath, resultBJsValue(resultBObject))
+              )
+            )
+        }
+    }
+  }
+
+  protected final def firstNameField(fields: Array[String]): String = fields(0)
+
+  protected final def lastNameField(fields: Array[String]): String = fields(1)
+
+  protected final def ninoField(fields: Array[String]): String = fields(2)
+
+  protected final val minChargeValueAllowed = BigDecimal("0.01")
+
+  protected final object MemberDetailsFieldNames {
+    val firstName = "firstName"
+    val lastName = "lastName"
+    val nino = "nino"
+  }
+
+  protected final def splitDayMonthYear(date: String): (String, String, String) = {
+    date.split("/").toSeq match {
+      case Seq(d, m, y) => Tuple3(d, m, y)
+      case Seq(d, m) => Tuple3(d, m, "")
+      case Seq(d) => Tuple3(d, "", "")
```

Comment:
should be throwing exception here instead of returning an empty String? or you are checking the empty string from the method caller?

Reply:
We don't want to throw an exception, since the field if it has no slashes in is invalid and will be shown as such when the validation errors are presented to the user. The else match should never happen since even if no slashes it will produce an array of one item. Probably the else could be removed - I put it in just in case.